### ISSUE TITLE
After upgrade GO to 1.7 version, the Code just won't show intellisense for a variable

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/vscode-go#441]
Moved from Microsoft/vscode#10639
---

From @Dmdv
- VSCode Version: Code 1.4.0 (6276dcb0ae497766056b4c09ea75be1d76a8b679, 2016-08-04T16:45:31.680Z)
- OS Version: Windows_NT ia32 10.0.10586

Steps to Reproduce:
1. choose GOLANG environment
2. start coding, declare import, for example fmt
3. after typing fmt. and pressing space there is no dropdown hint about available methods, just "Panic"

**Comment by sunnyawake5** [comment]
same issue here. fmt. doesn't work, fmt.Println( works.

**Comment by kkshetty** [comment]
Same here.

![screen shot 2016-08-18 at 5 47 27 pm](https://cloud.githubusercontent.com/assets/4770403/17773412/e954f5e0-656b-11e6-8bc4-1f276dd95814.png)

**Comment by lukehoban** [comment]
This looks like the same issue as #436 and #425.  Running `gocode close` should fix this.

**Comment by kkshetty** [comment]
```gocode close``` and a ```go get -u github.com/nsf/gocode``` fixed it for me.

Thanks @lukehoban

**Comment by asaf** [comment]
On MacOS, I had a gocode daemon running, I killed it / gocode close, re-opened studio and everything worked.

**Comment by tracer0tong** [comment]
Recipe works that's nice, but is it possible to include this gocode restart and update as part of new vscode-go? Or may be in documentation? At least this should be part of Release notes.

**Comment by Dmdv** [comment]
Recipe worked. Thank you

**Comment by lukehoban** [comment]
@tracer0tong I've updated the README with a note.  Unfortunately, I'm not sure it's safe/wise for the extension to restart the gocode daemon everything it launches, as that would have performance implications and potentially interact poorly with other tools that use gocode.

**Comment by hinst** [comment]
I get PANIC PANIC. I removed all tools, then installed all tools but it did not help

**Comment by lukehoban** [comment]
@hinst did you run `gocode close`

**Comment by utrescu** [comment]
**Fedora**: Only `gocode close` does not work. 

I need to do `gocode close` and `go get -u github.com/nsf/gocode` to get it working again.

**Comment by theGeekPirate** [comment]
```gocode close``` and ```go get -u github.com/nsf/gocode``` wasn't enough on Arch Linux, I also had to terminate the process manually.

**Comment by hilljgo** [comment]
Gocode will hog up my CPU every ten minutes or so, making my laptop sound like an airplane is about to take off. Having to run ```gocode close``` every ten minutes is quite annoying

**Comment by theGeekPirate** [comment]
@hilljgo Please search all previous issues to see if one is similar to your own problem (creating a new issue if not) instead of posting a completely irrelevant comment under this issue where it'll only get lost.

**Comment by WIZARD-CXY** [comment]
gocode close and go get -u github.com/nsf/gocode works

**Comment by Krognol** [comment]
I've tried `gocode close` and `go get -u github.com/nsf/gocode` but neither have fixed this problem for me.

Running Windows 10 64 bit
Go version 1.7.1

**Comment by kkshetty** [comment]
@Krognol Maybe ```gocode close``` isn't shutting down the daemon(as has been reported above in a few cases). Try searching for it and killing it manually. And then restart VS Code.

NOTE:- gocode's README instructs Windows users to do the below.

> Windows users should consider doing this instead:
> 
> ```go get -u -ldflags -H=windowsgui github.com/nsf/gocode```
> 
> That way on the Windows OS gocode will be built as a GUI application and doing so solves hanging window issues with some of the editors.

**Comment by qawarrior** [comment]
I was having this issue and I ran these commands
**go get -u github.com/nsf/gocode**
**gocode close**
**gocode -debug -s**
In that order and that fixed the issue

**Comment by Kleissner** [comment]
Not working for me as well. I tried everything from above (including deleting all directories in my src folder and redownloading all the dependencies) but it's still not working. It's just showing "PANIC" instead of the hints.

Windows 10 64-bit, Go 1.7

**Comment by IronPark** [comment]
if not work `gocode close`  for you
my solution
`killall -9 gocode`

**Comment by ramya-rao-a** [comment]
Closing this issue, as we have a workaround

**Comment by sfrooster** [comment]
I'm having an issue very similar to this except code completion works, but after a time vscode becomes unusable because there are godef processes hogging the CPU.  Killing them frees the cpu.

OSX Sierra, 1.7.1

**Comment by ramya-rao-a** [comment]
@sfrooster See #541 and #503

**Comment by qawarrior** [comment]
This seams to happen after every GO update go through the steps to kill the
barcode process to fix it

**Comment by dotnetjalps** [comment]
For me this is how it works.

Update the go version to Go 1.7.4 and then get latest version of all the tools.
go get -u -v github.com/nsf/gocode
go get -u -v github.com/rogpeppe/godef
go get -u -v github.com/golang/lint/golint
go get -u -v github.com/lukehoban/go-outline
go get -u -v sourcegraph.com/sqs/goreturns
go get -u -v golang.org/x/tools/cmd/gorename
go get -u -v github.com/tpng/gopkgs
go get -u -v github.com/newhook/go-symbols
go get -u -v golang.org/x/tools/cmd/guru

and Then restart the machine. It will work.

**Comment by kaitroll** [comment]
Help! 
go get -u -v github.com/rogpeppe/godef
# cd C:\Users\Ravend\Code\go\src\github.com\rogpeppe\godef; git config remote.origin.url
package github.com/rogpeppe/godef: remote origin not found

**Comment by gevgev** [comment]
Neither of the above solution has worked for me:

OS X El Capitan 10.11.6 (15G1217)
go version go1.7.5 darwin/amd64
Visual Studio Code Version 1.8.1 (1.8.1)

**Comment by ramya-rao-a** [comment]
@gevgev 
Did you try killing the current running gocode daemon as per https://github.com/Microsoft/vscode-go/issues/441#issuecomment-257314322
Once the gocode process is killed, run the command `Go: Install Tools` so that all dependent tools get updated

Are rest of the features of the Go extension working as expected?

**Comment by gevgev** [comment]
@ramya-rao-a 
Yes, I tried that several times, including with restarting my machine. 
And including installing all go tools manually
All other features, including all pop-ups, showing types of variable or even values of variables during debugging on mouse hover work just fine.
One thing I suspect is that with the latest Go lang Mac OS installation PKG the GOROOT now is set to "/usr/local/go", it is now kind of standardized with after I believe 1.7 release.
Which is why now I need to use sudo like in below scrip to install go tools, which means that Visual Studio Code simply cannot install the tools without sudo anymore, and maybe that might be the root cause:

[sudo -E -H gocode close
sudo -E -H go get -u -v github.com/nsf/gocode
sudo -E -H go get -u -v github.com/rogpeppe/godef
sudo -E -H go get -u -v github.com/golang/lint/golint
sudo -E -H go get -u -v github.com/lukehoban/go-outline
sudo -E -H go get -u -v sourcegraph.com/sqs/goreturns
sudo -E -H go get -u -v golang.org/x/tools/cmd/gorename
sudo -E -H go get -u -v github.com/tpng/gopkgs
sudo -E -H go get -u -v github.com/newhook/go-symbols
sudo -E -H go get -u -v golang.org/x/tools/cmd/guru](url)

**Comment by ramya-rao-a** [comment]
@gevgev I currently use go1.7.3, and I don't see the need to use sudo. Will update to go1.7.5 and get back.

Can you try running `gocode` from the command line to see if you get expected output?
If not, then we should take this up with `gocode`'s author

**Comment by gevgev** [comment]
Using `gocode` from command line:

Test file:
```
package main

import (
	"fmt"
)

func main() {
	fmt.
}
```

command line:
```
$ gocode -in="test.go" autocomplete c54
Found 1 candidates:
  PANIC PANIC PANIC
$ gocode -in="test.go" autocomplete c55
Found 1 candidates:
  PANIC PANIC PANIC
$ gocode -in="test.go" autocomplete c7
Nothing to complete.
$ gocode -in="test.go" autocomplete c8
Nothing to complete.
$ gocode -in="test.go" autocomplete c9
Found 1 candidates:
  func make(type, len[, cap]) type
$ gocode -in="test.go" autocomplete c10
Found 1 candidates:
  func make(type, len[, cap]) type
```
Per my understanding it should have suggested fmt.Print..Printf.. Println.. etc., but looks like the only suggestion gocode has is just 'make()' function, otherwise it is either 'Nothing to complete', or 'Panic'

Hope this helps.

**Comment by gevgev** [comment]
I have worked with nsf (author of gocode )on this issue, we got that resolved:
As it turns out I had a "leftover" stale instance (dated Sep 2016) sitting there!!:
```
$ which -a gocode
/Users/ggevorgyan/git/go/bin/gocode
/usr/local/go/bin/gocode
```
Removing the old instance and restarting gocode fixed it, everything works just fine now, including command line:
```
$ gocode -in="test.go" autocomplete c47
Found 25 candidates:
  func Errorf(format string, a ...interface{}) error
  func Fprint(w io.Writer, a ...interface{}) (n int, err error)
  func Fprintf(w io.Writer, format string, a ...interface{}) (n int, err error)
  func Fprintln(w io.Writer, a ...interface{}) (n int, err error)
  func Fscan(r io.Reader, a ...interface{}) (n int, err error)
  func Fscanf(r io.Reader, format string, a ...interface{}) (n int, err error)
  func Fscanln(r io.Reader, a ...interface{}) (n int, err error)
  func Print(a ...interface{}) (n int, err error)
  func Printf(format string, a ...interface{}) (n int, err error)
  func Println(a ...interface{}) (n int, err error)
  func Scan(a ...interface{}) (n int, err error)
  func Scanf(format string, a ...interface{}) (n int, err error)
  func Scanln(a ...interface{}) (n int, err error)
  func Sprint(a ...interface{}) string
  func Sprintf(format string, a ...interface{}) string
  func Sprintln(a ...interface{}) string
  func Sscan(str string, a ...interface{}) (n int, err error)
  func Sscanf(str string, format string, a ...interface{}) (n int, err error)
  func Sscanln(str string, a ...interface{}) (n int, err error)
  type Formatter interface
  type GoStringer interface
  type ScanState interface
  type Scanner interface
  type State interface
  type Stringer interface
```

Thank you!!!

**Comment by ramya-rao-a** [comment]
yay! glad you are unblocked @gevgev and thanks for sharing your solution.
Happy coding!

**Comment by freedomsean** [comment]
I can not use the function of autocomplete in VSCode Insider 1.11.0. Did I miss something? I have used the above solution, but it still doesn't have any suggestion.

> reinstall all reqired packages
> gocode close

If I use Code 1.10.1, it will be work again.

The following is my environment

VSCode Version: Code Insider 1.11.0
OS Version: Windows 7
Steps to Reproduce:

I use the same step to try it, but it still got error.

choose GOLANG environment
start coding, declare import, for example fmt
after typing fmt. and pressing space there is no dropdown hint about available methods or variables. There do not have any suggestion, even if "panic" .

**Comment by ramya-rao-a** [comment]
@freedomsean So on the same machine, autocomplete works fine in the stable build, but not insiders?

**Comment by freedomsean** [comment]
@ramya-rao-a  Yes, It works fine in the stable build, but not insiders. Did I need to submit this issue to VSCode project?

**Comment by ramya-rao-a** [comment]
@freedomsean I checked the insiders in Windows, everything is fine. But in Mac, none of the features of the Go extension is working. Can you share what platform you are on and if any of the other Go extension features (hover, go to defintion, add import command) are working for you or not?

**Comment by Russtopia** [comment]
Also check if you have copies of gocode.exe both in $GOROOT/bin (or %GOROOT%\bin on Windows) and $GOPATH/bin (%GOPATH%\bin) and be sure to remove all old versions before doing
<block>
go get -u github.com/nsf/gocode
</block>

**Comment by Twitch-CultCoder** [comment]
Perfect. I had the exact same problem on windows 7 Ultimate. Goclose and go get -u github.com/nsf/gocode fixed it for me. Thanks guys. Saved me huge headaches.

**Comment by scippio** [comment]
`gocode close and go get -u github.com/nsf/gocode` fix in insider: https://github.com/Microsoft/vscode/commit/89b158e11cb1c3fe94a3876222478ed2d0549fc8

**Comment by dc0d** [comment]
Happened today, in a strange way.

### How?
It did not work for a project (directory, opened with `code -g -n dir`). But worked for another project (Go 1.9.2, latest VSCode, Ubuntu 16.04).

I have:

* Updated all packages needed for the extension,
* Done `gocode stop` (and also `killall gocode` - both stopped the `gocode`),
* Done `gocode drop-cache` (no effect),
* And then ran `gocode -debug -s`.

My guess: This happens when number of candidates are more than one (in a sample case `Number of candidates found: 3`). There should be priority for searching for packages like:

1. `./internal/...`,
2. `./pkg/...`,
3. `./`,
4. `../.`,
5. `../up-to-$GOPATH/`,
4. Rest of `$GOPATH` (nearest neighbor? kidding!).

**Comment by dc0d** [comment]
@ramya-rao-a I've managed to reproduce it reliably.

When adding this comment with unicodes:

```
// 𐏃𐏂𐏁𐏀𐎿𐎾𐎽𐎼𐎻𐎺𐎹𐎸𐎷𐎶𐎵𐎴𐎳𐎲𐎱𐎰𐎯𐎮𐎭𐎬𐎫𐎪𐎩𐎨𐎧𐎦𐎥𐎤𐎣𐎢𐎡𐎠 := 1
```

It stops working and when this comment gets removed, it starts working.

**Comment by ramya-rao-a** [comment]
@dc0d I'd still prefer opening a new issue than a comment here. Rest of this issue either needed a restart of gocode or the issue with VS Code insiders. Both of which are unrelated to the detailed analysis you have done.

The multiple candidates is not an issue. It just means that there are multiple completion items.

I've logged #1431 to track the issue with unicode.
